stumpy/Mmotifs.py

```diff
@@ -0,0 +1,297 @@
+        The maximum amount of similar matches (nearest neighbors) of a motif
+        representative to be returned.
+        The first match is always the self-match for each motif.
+
+    max_distance: flaot, default None
+        Maximal distance that is allowed between a query subsequence
+        (a candidate motif) and all subsequences in T to be considered as a match.
+        If None, this defaults to
+        `np.nanmax([np.nanmean(D) - 2 * np.nanstd(D), np.nanmin(D)])`
+        (i.e. at least the closest match will be returned).
+
+    min_neighbors : int, default 1
+        The minimum number of similar matches a subsequence needs to have in order
+        to be considered a motif. This defaults to `1`, which means that a subsequence
+        must have at least one similar match in order to be considered a motif.
+
+    atol : float, default 1e-8
+        The absolute tolerance parameter. This value will be added to `max_distance`
+        when comparing distances between subsequences.
+
+    max_motifs: int, default 1
+        The maximum number of motifs to return
+
+    cutoff: numpy.ndarray or float, default None
+    The largest matrix profile value (distance) for each dimension of the
+    multidimensional matrix profile that a multidimenisonal candidate motif is
+    allowed to have.
+    If cutoff is only one value, these value will be applied to every dimension.
+
+    visualize_mdl: boolean, default 'False'
+        Visualize MDL results
+
+    Returns
+    -------
+    motif_matches_distances: numpy.ndarray
+        The distances corresponding to a set of subsequence matches for each motif.
+
+    motif_matches_indices: numpy.ndarray
+        The indices corresponding to a set of subsequences matches for each motif.
+
+    subspace: numpy.ndarray
+        A numpy.ndarray consisting of arrays that contain the `k`-dimensional
+        subspace for each motif.
+    """
+    if max_motifs < 1:  # pragma: no cover
+        logger.warning(
+            "The maximum number of motifs, `max_motifs`, "
+            "must be greater than or equal to 1"
+        )
+        logger.warning("`max_motifs` has been set to `1`")
+        max_motifs = 1
+
+    # Calculate subsequence length
+    T = _preprocess(T)  # convert dataframe to array if necessary
+    m = T.shape[-1] - P.shape[-1] + 1
+
+    # Calculate exclusion zone
+    excl_zone = int(np.ceil(m / config.STUMPY_EXCL_ZONE_DENOM))
+
+    if max_matches is None:
+        max_matches = np.inf
+
+    motif_matches_distances = []
+    motif_matches_indices = []
+    subspace = []
+
+    # Identify the first multidimensional motif as candidate motif and get its
+    # nearest neighbor
+    candidate_idx = np.argsort(P, axis=1)[:, 0]
+    nn_idx = I[np.arange(len(candidate_idx)), candidate_idx]
+
+    # Iterate over each motif
+    while len(motif_matches_distances) < max_motifs:
+
+        # Choose dimension k using MDL
+        mdl, subspaces = stumpy.mdl(T, m, candidate_idx, nn_idx)

```

Comment:
to be consistent with the multidimensional tutorial, let's do:

`mdls, subspaces = mdl(T, m, candidate_idx, nn_idx)`

Reply:
I don't know how to do it without writing `stumpy.mdl` since `from . import mdl` doesn't work. Currently, however, it is no different in the tutorial, I think.

Reply:
> since from . import mdl doesn't work

Can you please elaborate on what you mean by this? What doesn't work? How are testing whether or not it works? I should be able to help, I just need a little bit more information to help you debug.


> Currently, however, it is no different in the tutorial, I think.

So, the Python notation `from . import something` is specific to when you are writing code for a Python package and is a "relative import" to STUMPY. Within a package we should avoid doing `from stumpy import something`.

After writing/changing your source code, you should always remember to reinstall STUMPY with:

```
./setup.sh
```

However, before you import STUMPY for testing, you need to navigate away from the STUMPY directory first (say, to your desktop directory)! Then, you can try:

```
from stumpy.mmotifs import import mmotifs 
```